src/text/writer.rs

```diff
@@ -0,0 +1,352 @@
+use crate::result::{illegal_operation, IonResult};
+use crate::text::encoder::TextEncoder;
+use crate::IonType;
+use bigdecimal::BigDecimal;
+use chrono::{DateTime, FixedOffset};
+use std::io::{BufWriter, Write};
+
+pub struct TextWriter<W: Write> {
+    output: BufWriter<W>,
+    annotations: Vec<String>,
+    field_name: Option<String>,
+    containers: Vec<IonType>,
```

Comment:
Interesting, so this type copies in these concerns?  What is the purpose of the containers `Vec`? Is that the current stack of where we "are" in the writing?

Reply:
> Is that the current stack of where we "are" in the writing?

Yes. It's being used to track which value delimiters to insert between child values (`,`/`\n`/` `) and whether a field name is needed.

>  so this type copies in these concerns? 

Sorry, unsure what you mean here. Could you rephrase?

Reply:
Sorry, "this type stores copies of the working fields and annotations" is what I meant.